webclient/app/src/app/commons/appTemplateCard/AppTemplateCard.jsx

```diff
@@ -0,0 +1,128 @@
+import React, { useState } from "react";
+import { styled } from '@mui/material/styles';
+import { Container, Collapse, Card, CardContent, CardActions, List, ListItem, ListItemIcon, ListItemText, Typography, Divider, Grid, IconButton } from "@mui/material";
+import AppTemplateCardStyles from "./AppTemplateCardStyles";
+import { useTranslation } from "react-i18next";
+import GitHub from '@mui/icons-material/GitHub';
+import { Delete, Edit } from "@mui/icons-material";
+import SyntaxHighlighter from "react-syntax-highlighter/dist/esm/light";
+import js from 'react-syntax-highlighter/dist/esm/languages/hljs/javascript';
+import dark from 'react-syntax-highlighter/dist/esm/styles/hljs/dark';
+import ExpandMoreIcon from '@mui/icons-material/ExpandMore';
+import AppTemplateDialog from "../../commons/appTemplateDialog/AppTemplateDialog";
+import AppTemplateRest from "../../services/AppTemplateRest"
+import ConfirmationDialog from "../alert/ConfirmationDialog";
+
+const ExpandMore = styled((props) => {
+    const { expand, ...other } = props;
+    return <IconButton {...other} />;
+})(({ theme, expand }) => ({
+    transform: !expand ? 'rotate(0deg)' : 'rotate(180deg)',
+    marginLeft: 'auto',
+    transition: theme.transitions.create('transform', {
+        duration: theme.transitions.duration.shortest,
+    }),
+}));
+
+
+function AppTemplateCard(props) {
+
+    SyntaxHighlighter.registerLanguage('javascript', js);
+
+    const { appTemplate, handleRefresh } = props;
+    const appTemplateCardStyles = AppTemplateCardStyles();
+    const { t } = useTranslation();
+    const [expanded, setExpanded] = React.useState(false);
+    const [selectedAppTemplate, setSelectedAppTemplate] = useState(false);
+    const [openDialog, setOpenDialog] = React.useState(false);
+    const appTemplateRest = new AppTemplateRest();
+
+    const [openDeleteDialog, setOpenDeleteDialog] = useState(false);
+    const deleteDialolgContent = ({ "title": "appTemplateDeleteDialog.title", "message": "appTemplateDeleteDialog.message" });
+
+    const closeDeleteDialog = () => {
+        setOpenDeleteDialog(false);
+    }
+
+    const handleExpandClick = () => {
+        setExpanded(!expanded);
+    };
+
+    const handleDialogOpen = () => {
+        setOpenDialog(true);
+        setSelectedAppTemplate(appTemplate);
+    };
+
+    const handleDialogClose = () => {
+        setOpenDialog(false);
+    };
+
+    const  handleDelete = (appTemplateId) => {
```

Comment:
This should be handled by the parent component

Reply:
Why?